Mathlib/LinearAlgebra/FreeAlgebra.lean

```diff
@@ -22,21 +22,28 @@ universe u v
 
 namespace FreeAlgebra
 
+variable (R : Type u) (X : Type v)
+
+section
+variable [CommSemiring R]
+
 /-- The `FreeMonoid X` basis on the `FreeAlgebra R X`,
 mapping `[x₁, x₂, ..., xₙ]` to the "monomial" `1 • x₁ * x₂ * ⋯ * xₙ` -/
 -- @[simps]
-noncomputable def basisFreeMonoid (R : Type u) (X : Type v) [CommRing R] :
-    Basis (FreeMonoid X) R (FreeAlgebra R X) :=
+noncomputable def basisFreeMonoid : Basis (FreeMonoid X) R (FreeAlgebra R X) :=
   Finsupp.basisSingleOne.map (equivMonoidAlgebraFreeMonoid (R := R) (X := X)).symm.toLinearEquiv
 #align free_algebra.basis_free_monoid FreeAlgebra.basisFreeMonoid
 
--- TODO: generalize to `X : Type v`
-theorem rank_eq {K : Type u} {X : Type max u v} [Field K] :
-    Module.rank K (FreeAlgebra K X) = Cardinal.mk (List X) :=
-  -- Porting note: the type class inference was no longer automatic.
-  -- was: (Cardinal.lift_inj.mp (basisFreeMonoid K X).mk_eq_rank).symm
-  Cardinal.lift_inj.mp (@Basis.mk_eq_rank _ _ _ _ _ _ (inferInstance : Module K (FreeAlgebra K X))
-  (basisFreeMonoid K X)).symm
+instance : Module.Free R (FreeAlgebra R X) :=
+  have : Module.Free R (MonoidAlgebra R (FreeMonoid X)) := Module.Free.finsupp _ _ _
+  Module.Free.of_equiv (equivMonoidAlgebraFreeMonoid (R := R) (X := X)).symm.toLinearEquiv
+
+end
+
+theorem rank_eq [CommRing R] [StrongRankCondition R] :
```

Comment:
You should never need to mention `StrongRankCondition` is the commutative setting because of [commRing_strongRankCondition](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/FreeModule/StrongRankCondition.html#commRing_strongRankCondition)
```suggestion
theorem rank_eq [CommRing R] :
```
I guess you might also need to invoke `nontriviality R` below to make this work.

Reply:
I'm aware of that, but I think in practice almost nothing imports that file because its dependencies are too heacy, so we have lots of redundant `[StrongRankCondition R]`s across mathlib. Do you want me to add the import here?

Reply:
Yes, I think we should take the import hit.

Mathematically it's very weird to say a ring is commutative and satisfies the strong rank condition. I wasn't aware we had followed this pattern elsewhere and I would advocate undoing it elsewhere.

Reply:
> I guess you might also need to invoke `nontriviality R` below to make this work.

Neither result is true for trivial `R`, so I've replaced `StrongRankCondition` with `Nontrivial`.